cron/appstats.go

```diff
@@ -13,34 +14,61 @@ import (
 
 	_ "github.com/lib/pq"
 )
+type App struct {
+	Name	string
+	ID		string
+	Count	int
+}
+
 
-const (
-	host	= "trotter.cyverse.org"
-	port   	= 6432
-	user   	= "de"
-	dbname 	= "de"
-	folder  = "/tmp"
-	logfile = "logs-stdout-output"
-)
-func main(){
-	psqlInfo := fmt.Sprintf("host=%s port=%d user=%s "+
-		"dbname=%s sslmode=disable", host, port, user, dbname)
-	db, err := sql.Open("postgres", psqlInfo)
+
+func GetTopApps(db *sql.DB, amount int, days int) []App{
+	var name *string
+	var appID string
+	var appCount int
+
+	query := `SELECT app_name, app_id, count(*) AS job_count FROM jobs
+           WHERE start_date >= (now() - ($2 || ' DAY')::INTERVAL )
+           AND app_id != '1e8f719b-0452-4d39-a2f3-8714793ee3e6'
+           GROUP BY app_name, app_id
+           ORDER BY job_count DESC
+           LIMIT $1`
+	rows, err := db.Query(query, amount, days)
 
 	if err != nil {
 		panic(err)
 	}
 
-	defer db.Close()
+	defer rows.Close()
+
+	apps := make([]App, amount)
+	for i := 0; rows.Next(); i++{
+		err := rows.Scan(&name, &appID, &appCount)
+		if err != nil {
+			panic(err)
```

Comment:
`panic` will actually cause the program to abort unless it's recovered. (See https://blog.golang.org/defer-panic-and-recover for details). You'll probably want to have `GetTopApps` return both a lit of apps and an error instead. Then the function that calls `GetTopApps` can check the error return value for possible errors.

Reply:
So instead of panic it should return err and the current state of the array?

Reply:
Yep.